### PR TITLE
k8s-int-test-build: zk-less druid cluster and http based segment/task managment

### PR DESCRIPTION
**Pull request by himanshug**
Linked #10542

### Description

Follow up on #10669

- Updates K8S based integration test build to deploy zk-less Druid cluster along with http based segment and remote task management to sanity test #10544

- Splits https://github.com/apache/druid/blob/1884c356984695cf31ef63dd0075602b5087ec4b/integration-tests/script/build_run_k8s_cluster.sh into 3 separate scripts , "setup_k8s_cluster.sh", "setup_druid_operator_on_k8s.sh" and "setup_druid_on_k8s.sh" for easily running specific parts while debugging setup or test failures locally.

- Updates to k8s extension documentation for few gotchas.

### CODE REVIEW

File: docs/development/extensions-core/kubernetes.md [hunk 1]
```diff
@@ -57,3 +55,34 @@ Additionally, this extension has following configuration.
 |`druid.discovery.k8s.renewDeadline`|`Duration`|Lease renewal period used by Leader.|PT17S|No|
 |`druid.discovery.k8s.retryPeriod`|`Duration`|Retry wait used by Leader Election algorithm on failed operations.|PT5S|No|
 
+### Gotchas
+
+- Label/Annotation path in each pod spec MUST EXIST, which is easily satisfied if there is at least one label/annotation in the pod spec already. This limitation may be removed in future.
+- Druid Pods need permissions to be able to add labels to self-pod, List and Watch other Pods, create ConfigMap for leader election. Assuming, "default" service account is used by Druid pods, you might need to add following or something similar Kubernetes Role and Role Binding.
```
Comment (zhangyue19921010) [inline docs/development/extensions-core/kubernetes.md:61]: Just curious. Which kind of druid server needs these `add labels to self-pod, List and Watch other Pods, create ConfigMap` permissions? MiddleManager or all the druid servers.

File: integration-tests/script/setup_druid_on_k8s.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+$KUBECTL apply -f integration-tests/k8s/tiny-cluster.yaml
+
+# Wait 4 * 15 seconds to launch pods.
+#count=0
+#JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until sudo /usr/local/bin/kubectl -n default get pods -lapp=travis-example -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 4;if [ $count -eq 15 ];then break 2 ;else let "count++";fi;echo $i;echo "waiting for travis-example deployment to be available"; sudo /usr/local/bin/kubectl get pods -n default; done
+sleep 120
+
+## Debug And FastFail
+
+sudo /usr/local/bin/kubectl get pod
```
Comment (zhangyue19921010) [inline integration-tests/script/setup_druid_on_k8s.sh:52]: nit: `/usr/local/bin/kubectl` can be replaced by `$KUBECTL`. The same as `sudo /usr/local/bin/kubectl get svc`

File: integration-tests/script/setup_k8s_cluster.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+set -e
+
+# Set Necessary ENV
+export CHANGE_MINIKUBE_NONE_USER=true
+export MINIKUBE_WANTUPDATENOTIFICATION=false
+export MINIKUBE_WANTREPORTERRORPROMPT=false
+export MINIKUBE_HOME=$HOME
+export KUBECONFIG=$HOME/.kube/config
+export KUBECTL="sudo /usr/local/bin/kubectl"
```
Comment (zhangyue19921010) [inline integration-tests/script/setup_k8s_cluster.sh:25]: nit: it seems that `export KUBECTL="sudo /usr/local/bin/kubectl"` is never be used in `setup_k8s_cluster.sh `. Maybe can be removed?

File: docs/development/extensions-core/kubernetes.md (hunk 1 above)
Comment (himanshug) [reply docs/development/extensions-core/kubernetes.md:61]: everyone needs pod list/update .
`create/update ConfigMap` is needed only by coordinators/overlords which participate in leader election ... but `read ConfigMap` is needed by all so as to know who the leader is. That said, I haven't tested such granularity and would just give same permission to all pods.

File: integration-tests/script/setup_druid_on_k8s.sh (hunk 2 above)
Comment (himanshug) [reply integration-tests/script/setup_druid_on_k8s.sh]: changed

File: integration-tests/script/setup_k8s_cluster.sh (hunk 3 above)
Comment (himanshug) [reply integration-tests/script/setup_k8s_cluster.sh]: changed

**Comment by himanshug** [comment]
@zhangyue19921010 thanks for taking a look

**Review by zhangyue19921010 — APPROVED** [review]

**Comment by zhangyue19921010** [comment]
LTGM :)

**Review by clintropolis — APPROVED** [review]
🎉 

As a follow-up we should try to get some or all of https://github.com/apache/druid/blob/master/integration-tests/src/test/java/org/apache/druid/tests/leadership/ITHighAvailabilityTest.java running on this configuration.

**Comment by zhangyue19921010** [comment]
> As a follow-up we should try to get some or all of https://github.com/apache/druid/blob/master/integration-tests/src/test/java/org/apache/druid/tests/leadership/ITHighAvailabilityTest.java running on this configuration.

Hi @clintropolis and @himanshug  Are there any progress being made on that? If no, I'd like to take the chance to implement it after this PR have been merged. :)

**Comment by himanshug** [comment]
@clintropolis thanks for approving, I will do one more round of checks of things in this PR tomorrow and merge.

> Are there any progress being made on that? If no, I'd like to take the chance to implement it after this PR have been merged

@zhangyue19921010 I did some work there and will finish it, I am adding few more related things in that PR . that works currently lives in my fork at https://github.com/himanshug/druid/tree/k8s_build2

**Pull request merged**